### ISSUE TITLE
template conversion for vmware provissioner. 

### ISSUE THREAD

**Issue by athingisathing** [mitchellh/packer#2918]
I am testing some code to convert a vm to a template but im new to go.... it's erroring but its not outputting the error. what am I missing?

```go
	targs := []string{
		fmt.Sprintf("--url https://%s:443/sdk/webService", p.config.Host),
		fmt.Sprintf("--username %s@visa.com", p.config.Username),
		fmt.Sprintf("--password '%s'", p.config.Password),
		fmt.Sprintf("--vmname %s", p.config.VMName),
		"--operation T",
	}

	ui.Message(fmt.Sprintf("Uploading %s to vSphere", vmx))
	var out bytes.Buffer
	log.Printf("Starting ovftool with parameters: %s", strings.Join(args, " "))
	cmd := exec.Command("ovftool", args...)
	cmd.Stdout = &out
	if err := cmd.Run(); err != nil {
		return nil, false, fmt.Errorf("Failed: %s\nStdout: %s", err, out.String())
	}
    var tout bytes.Buffer
	ui.Message(fmt.Sprintf("Converting %s to Template", p.config.VMName))
	tcmd := exec.Command("/usr/lib/vmware-vcli/apps/vm/vmtemplate.pl", targs...)
	tcmd.Stdout = &tout
	if err := tcmd.Run(); err != nil {
		return nil, false, fmt.Errorf("Failed: %s\nStdout: %s", err, tout.String())
	}
```

**Comment by athingisathing** [comment]
do i need a new error object?

also is there any way to test a command over and over or do i need to compile and test compile test....

**Comment by cbednarski** [comment]
Sorry to keep you waiting on this! A lot of packer functionality (such as builders, provisioners, and post-processors) run in a separate process which is started by the main packer executable. When the plugin starts up the main process attaches to the plugin's stdout and stderr and captures the output, so if you print from the plugin you won't see anything in your terminal. `log.Printf` always works so I use that for debugging.

> also is there any way to test a command over and over or do i need to compile and test compile test....

If you write your new feature in a unit test you can use `go test ./path/to/folder/...` to run things more quickly. When you want to test against a packer template you will need to run `make dev` and recompile the binary. This should be faster now since there is only one binary to compile for all of the plugins (and go 1.4 is even faster than 1.5)

There is an example of a unit test in https://github.com/mitchellh/packer/blob/master/post-processor/compress/post-processor_test.go You would run this via `go test ./post-processor/compress/...`.

Please feel free to open a new issue if you have any other questions. Thanks for hacking on packer!
